AppFramework/Core/GREYElementInteraction.m

```diff
@@ -802,13 +803,23 @@ - (BOOL)grey_handleFailureOfAssertion:(id<GREYAssertion>)assertion
 
 - (NSError *)grey_errorToReturnForInteractionError:(GREYError *)interactionError
                                         withReason:(NSString *)reason {
+  // TODO: @wsaid remove the `reason` parameter and replace it with the errorDetails dict
```

Comment:
//TODO(wsaid): Remove the reason parameter and replace with the errorDetails dictionary.